packages/components/src/keyboard-shortcuts/test/index.tsx

```diff
@@ -6,13 +6,20 @@ import { fireEvent, render, screen } from '@testing-library/react';
 /**
  * Internal dependencies
  */
-import KeyboardShortcuts from '../';
+import KeyboardShortcuts from '..';
 
 describe( 'KeyboardShortcuts', () => {
-	function keyPress( which, target ) {
+	function keyPress(
+		which: KeyboardEvent[ 'which' ],
+		target: Parameters< typeof fireEvent >[ 0 ]
+	) {
 		[ 'keydown', 'keypress', 'keyup' ].forEach( ( eventName ) => {
-			const event = new window.Event( eventName, { bubbles: true } );
+			const event = new window.Event( eventName, {
+				bubbles: true,
+			} );
+			// @ts-expect-error
 			event.keyCode = which;
+			// @ts-expect-error
 			event.which = which;
```

Comment:
I think these are inevitable, because `fireEvent()` requires the event argument to be of type `Event` and not the more specific `KeyboardEvent`. I'm happy to be wrong though.

Reply:
I tried using `createEvent` locally which seems to work and circumvent the TS error:

```tsx
function keyPress(
		which: number,
		target: Parameters< typeof fireEvent >[ 0 ]
	) {
		[ 'keydown', 'keypress', 'keyup' ].forEach( ( eventName ) => {
			const event = createEvent(
				eventName,
				target,
				{
					bubbles: true,
					keyCode: which,
					which,
				},
				{ EventType: 'KeyboardEvent' }
			);
			fireEvent( target, event );
		} );
	}
```

Although if you'd assign `event.which` after the event was created it'd still warn that `which` (or `keyCode`) doesn't exist on `Event`.

Reply:
TIL, thank you!